packages/flutter/test/widgets/sliver_cross_axis_group_test.dart

```diff
@@ -511,6 +511,60 @@ void main() {
     expect(first.localToGlobal(Offset.zero), Offset.zero);
     expect(second.localToGlobal(Offset.zero), const Offset(VIEWPORT_WIDTH / 2, 0));
   });
+
```

Comment:
There are a lot of configurations of SliverPersistentHeader we should test here. SliverAppBar is a SliverPersistentHeader with some extra features. That should be tested too.

Here floating: true (rest false) and pinned:true (rest false) are covered, but can you also add tests for:

- floating: false, pinned: false, snap: false
- floating: true, pinned: false, snap: true
- floating: true, pinned: true, snap: false
- floating: true, pinned: true, snap: true

(snap can only be true when floating is true, so that combination is irrelevant)

Reply:
Done! For the floating: true, pinned: true, snap: false, I used SliverPersistentHeader(floating: true, pinned: true) instead since I think this is the same cas.e